### PR TITLE
feat(lint): add unused, and then replace with megacheck

### PR DESCRIPTION
**Pull request by Jimexist**
**What this PR does**
add `unused` check and then replace with metacheck

**why we need it**

**Special notes for your reviewer**:
why those code were not used?

`megacheck` = `unused` + `staticcheck` + `gosimple`

### CODE REVIEW

**Comment by ddysher** [comment]
@zoumo to take a look

File: cmd/flag-gen/types/types.go [hunk 1]
```diff
@@ -16,27 +16,24 @@ limitations under the License.
 
 package types
 
-import (
-	"time"
-)
-
 // go:generate flag-gen -i github.com/caicloud/nirvana/cmd/flag-gen/types -o github.com/caicloud/nirvana/cli
 
+// TODO: why not used?
```
Comment (zoumo) [inline cmd/flag-gen/types/types.go]: flag-gen read this struct to generate code

File: cli/config_test.go [hunk 2]
```diff
@@ -91,9 +91,10 @@ func TestIsSet(t *testing.T) {
 }
 
 func TestGet(t *testing.T) {
-	type args struct {
-		key string
-	}
+	// TODO: why not used?
```
Comment (zoumo) [inline cli/config_test.go]: auto generated by vscode plugin, can be deleted

File: cmd/flag-gen/types/types.go (hunk 1 above)
Comment (ddysher) [reply cmd/flag-gen/types/types.go]: then we have to trick the tool to not warning on this?

Comment (Jimexist) [reply cmd/flag-gen/types/types.go]: fixed by adding `// nolint:`

File: cli/config_test.go (hunk 2 above)
Comment (Jimexist) [reply cli/config_test.go]: removed

**Pull request merged**